src/datastore/helpers.ts

```diff
@@ -1343,14 +1343,26 @@ export function newReOrgUpdatedEntities(): ReOrgUpdatedEntities {
  */
 export class PgWriteQueue {
   readonly queue: PQueue;
+  private tasks: Promise<unknown>[];
   constructor() {
     const concurrency = Math.max(1, getUintEnvOrDefault('STACKS_BLOCK_DATA_INSERT_CONCURRENCY', 4));
     this.queue = new PQueue({ concurrency, autoStart: true });
+    this.tasks = [];
   }
   enqueue(task: Parameters<PQueue['add']>[0]): void {
-    void this.queue.add(task);
+    const p = this.queue.add(task);
+    p.catch(e => logger.error(e, 'PgWriteQueue task failed'));
```

Comment:
If there's no catch, the process will be terminated by an unhandledRejection.

The error is logged here because `throw firstRejected.reason` only throws the first one, we don't have any log for the other errors.

Reply:
Ah okay, would it make sense to use the p-queue error event instead? https://www.npmjs.com/package/p-queue#error

Reply:
This is what we saw using a node built from this PR.

![image](https://github.com/hirosystems/stacks-blockchain-api/assets/432106/12cc0759-654a-4715-b28c-681501f1966f)

Reply:
I see, we can manually capture the errors ourself with `.catch` to prevent a `terminated by an unhandledRejection`. But could se instead use the build-in p-queue errror handler, e.g. 
```
queue.on('error', error => {
	console.error(error);
});
```
?

Reply:
This is not available before v7.0.0: https://github.com/sindresorhus/p-queue/tree/v6.6.2?tab=readme-ov-file#events